mbtiles.go

```diff
@@ -92,14 +119,48 @@ func Open(path string) (*MBtiles, error) {
 	db := &MBtiles{
 		filename:  path,
 		pool:      pool,
-		timestamp: stat.ModTime().Round(time.Second),
+		timestamp: modTime,
 		format:    format,
 		tilesize:  tilesize,
 	}
 
 	return db, nil
 }
 
+func fileModTime(path string) (time.Time, error) {
+	stat, err := os.Stat(path)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return time.Time{}, fmt.Errorf("path does not exist: %q", path)
+		}
+		return time.Time{}, err
+	}
+	return stat.ModTime().Round(time.Second), nil
+}
+
+func validateAndGetFormatAndSize(path string) (TileFormat, uint32, error) {
+	// there must not be a corresponding *-journal file (tileset is still being created)
```

Comment:
Suggest moving the check for absence of the `-journal` file to `getModTime`, since we don't care about mod time if we reject it because it is still being updated.  That keeps the file path related operation together.